src/GraphQLRequestFactory.php

```diff
@@ -26,13 +28,14 @@ public function __construct(RequestFactoryInterface $requestFactory, StreamFacto
     /**
      * Creates POST Request with application/json Content-Type header
      *
-     * @param array<string, mixed> $variables
+     * @param StreamInterface|string $query
+     * @param array<string, mixed>   $variables
      */
-    public function createRequest(string $uri, string $query, ?array $variables = null, ?string $operationName = null) : RequestInterface
+    public function createRequest(string $uri, $query, ?array $variables = null, ?string $operationName = null) : RequestInterface
     {
         $request = $this->requestFactory->createRequest('POST', $uri);
 
-        $body = ['query' => $query];
+        $body = ['query' => is_string($query) ? $query : (string) $query];
```

Comment:
bere to nejen steamInterface a string, ale cokoliv co se umí transformovat na string ... neříkám že je to issue, jen aby ti to omylem neuteklo...

Reply:
nj, je to otypovany, tak pohoda. S PHP8 prijde concrete union type